src/libelfmaster.c

```diff
@@ -2297,3 +2298,351 @@ elf_close_object(elfobj_t *obj)
 	 */
 	munmap(obj->mem, obj->size);
 }
+
+bool
+elf_has_header(const char *path, bool *has_header, elf_error_t *error) 
+{
+ 	int fd;
+        int magic;
+
+        if((fd = open(path, O_RDONLY)) < 0) {
+		elf_error_set(error, "open: %s", strerror(errno));
+		return false;
+	}
+	
+	if(pread(fd, &magic, sizeof(int), 0) != sizeof(int)) {
+		elf_error_set(error, "pread: %s", strerror(errno));
+		return false;
+	}
+	*has_header = (magic == 0x464c457f) ? true : false;
+	close(fd);
+	return true;
+}
+
+/*
+ * Handles opening binary stubs that reside in disk. Saves instance as a elfobj_t.
+ */
+bool
+elf_open_stub(const char *path, struct elfobj *obj, elf_error_t *error) 
+{
+	int fd;
+	struct stat st;
+	unsigned int open_flags = O_RDWR;
+	unsigned int mmap_perms = PROT_READ|PROT_WRITE;
+	unsigned int mmap_flags = MAP_PRIVATE;
+	uint8_t *mem;
+
+	memset(obj, 0, sizeof(*obj));
+	if((fd = open(path, open_flags, S_IRUSR|S_IRGRP|S_IROTH)) < 0) {
+		elf_error_set(error, "open: %s", strerror(errno));
+		return false;
+	}
+	obj->fd = fd;
+	obj->path = path;
+		
+	if (fstat(fd, &st) < 0) {
+		elf_error_set(error, "fstat: %s", strerror(errno));
+		close(fd);
+		return false;
+	}
+	obj->size = st.st_size;
+
+	if((mem = mmap(NULL, st.st_size, mmap_perms, mmap_flags, fd, 0)) == MAP_FAILED) {
+		elf_error_set(error, "mmap: %s", strerror(errno));
+		close(fd);
+		return false;
+	}
+	obj->mem = mem;
+	obj->type = ELF_ET_STUB;
+	return true;
+}
+
+/*
+ * Handles initialization of a elfobj_t for a given stub supplied as a binary array.
+ */
+bool
+elf_init_stub(struct elfobj *obj, uint8_t *mem, size_t size, elf_error_t *error) 
+{
+	if (obj == NULL) {
+		elf_error_set(error, "Invalid elfobj_t address");
+		return false;
+	}
+	if (mem == NULL) {
+		elf_error_set(error, "Invalid stub buffer address");
+		return false;
+	}
+	obj->type = ELF_ET_STUB;
+	obj->mem = mem;
+	obj->size = size;
+	return true;
+}
+
+/*
+ * Creates an elf object. initial content can be specified to contain a given ELF file.
+ */
+bool
+elf_create_object(const char *path, struct elfobj *obj, struct elfobj *copy, size_t size, uint64_t load_flags, elf_error_t *error) 
```

Comment:
cut off before 80 chars, then when you continue on the next line do 4 spaces i.e.
```
bool some_function(char *test, int arg2, int arg3,
    struct test *tp) 
{```